### PR TITLE
fix(groups): Hide group profile fields that don't have a value

### PR DESCRIPTION
**Pull request by juho-jaakkola**

### CODE REVIEW

File: mod/groups/actions/groups/edit.php [hunk 1]
```diff
@@ -19,6 +19,14 @@ function profile_array_decoder(&$v) {
 foreach (elgg_get_config('group') as $shortname => $valuetype) {
 	$input[$shortname] = get_input($shortname);
 
+	if ($input[$shortname] === '') {
+		// The group profile displays all profile fields that have a value.
+		// We don't want to display fields with empty string value, so we
+		// prevent it getting saved by setting the value to null.
+		$input[$shortname] = null;
```
Comment (mrclay) [inline mod/groups/actions/groups/edit.php]: Does setting to null actually remove metadata?

Comment (mrclay) [inline mod/groups/actions/groups/edit.php]: I think we just handle this where we copy $input to metadata. If it's empty string, deleteMetadata() instead.

Comment (juho-jaakkola) [inline mod/groups/actions/groups/edit.php]: ElggEntity in fact first removes all metadata: https://github.com/Elgg/Elgg/blob/1.10/engine/classes/ElggEntity.php#L412

Metadata service later ignores the null value: https://github.com/Elgg/Elgg/blob/1.10/engine/classes/Elgg/Database/MetadataTable.php#L123

Comment (mrclay) [inline mod/groups/actions/groups/edit.php]: What do you think about using delete or unset() until #8042 is resolved?

Comment (juho-jaakkola) [inline mod/groups/actions/groups/edit.php]: Sure

**Comment by juho-jaakkola** [comment]
Updated to use ```deleteMetadata()```

**Comment by mrclay** [comment]
LGTM

**Pull request merged**